tests/apps/search/test_indexers_courses.py

```diff
@@ -356,6 +358,70 @@ def test_indexers_courses_get_es_documents_no_end_no_enrollment_end(self):
             indexed_courses[0]["course_runs"][0]["enrollment_end"].year, 9999
         )
 
+    def test_indexers_courses_get_es_document_no_image_cover_picture(self):
+        """
+        ES document is created withot errors when a cover image for the course is
+        actually a Picture instance without an image on it.
+        """
+        # Create the example course to index
+        course = CourseFactory(should_publish=True)
+        # And a picture with no image to make sure the indexer does not break
+        cover_picture = PictureFactory(picture=None)
+        # Make sure we associate the picture with the course through the cover placeholder
+        cover_picture.cmsplugin_ptr.placeholder = (
+            course.extended_object.get_public_object()
+            .placeholders.filter(slot="course_cover")
+            .first()
+        )
+        cover_picture.cmsplugin_ptr.save()
+
+        indexed_courses = list(
+            CoursesIndexer.get_es_documents(index="some_index", action="some_action")
+        )
+
+        self.assertEqual(len(indexed_courses), 1)
+        self.assertEqual(
+            indexed_courses[0]["_id"],
+            str(course.extended_object.get_public_object().id),
+        )
+        self.assertEqual(indexed_courses[0]["cover_image"], {})
+
+    def test_indexers_courses_get_es_document_no_image_icon_picture(self):
+        """
+        ES document is created withot errors when a icon image for the course is
+        actually a Picture instance without an image on it.
+        """
+        # Create the example course to index
+        course = CourseFactory(should_publish=True)
+        course_icons_placeholder = (
+            course.extended_object.get_public_object()
+            .placeholders.filter(slot="course_icons")
+            .first()
+        )
+        # And a picture with no image to make sure the indexer does not break
+        cover_picture = PictureFactory(picture=None)
+        # Make sure we associate the picture with the course through the icons placeholder
+        cmsplugin_ptr = CMSPlugin()
```

Comment:
@sampaccoud this is the test I'm having trouble with. I need both a `Picture` and a `CategoryPluginModel ` instances with the same `CMSPlugin` on them (or at least something that will match on the queries for icons on the indexer).

As you can see above I managed it with the cover image query but there was no `CategoryPluginModel` there.

As a small aside, I'm also wondering how in one of the icon queries the placeholder slot is named `icon` and in the other it's `course_icons. Maybe I missed something there.